app/api/speakers.py

```diff
@@ -0,0 +1,92 @@
+
+from app.api.helpers.utilities import dasherize
+from app.api.helpers.permissions import jwt_required
+from app.models import db
+from app.models.speaker import Speaker
+from app.models.event import Event
+
+
+class SpeakerSchema(Schema):
+    class Meta:
+        type_ = 'speaker'
+        self_view = 'v1.speaker_detail'
+        self_view_kwargs = {'id': '<id>'}
+        inflect = dasherize
+
+    id = fields.Str(dump_only=True)
+    name = fields.Str(required=True)
+    photo = fields.Str()
+    thumbnail = fields.Str()
+    small = fields.Str()
+    icon = fields.Str()
+    short_biography = fields.Str()
+    long_biography = fields.Str()
+    speaking_experience = fields.Str()
+    email = fields.Str(required=True)
+    mobile = fields.Str()
+    website = fields.Str()
+    twitter = fields.Str()
+    facebook = fields.Str()
+    github = fields.Str()
+    linkedin = fields.Str()
+    organisation = fields.Str()
+    featured = fields.Boolean(default=False)
+    position = fields.Str()
+    country = fields.Str()
+    city = fields.Str()
+    gender = fields.Str()
+    heard_from = fields.Str()
+    sponsorship_required = fields.Str()
+    event = Relationship(attribute='event',
+                         self_view='v1.speaker_event',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.event_detail',
+                         related_view_kwargs={'speaker_id': '<id>'},
+                         schema='EventSchema',
+                         type_='event')
+    user = Relationship(attribute='user',
+                        self_view='v1.speaker_user',
+                        self_view_kwargs={'id': '<id>'},
+                        related_view='v1.user_detail',
+                        related_view_kwargs={'speaker_id': '<id>'},
+                        schema='UserSchema',
+                        type_='user')
+
+
+class SpeakerList(ResourceList):
+    def query(self, view_kwargs):
+        query_ = self.session.query(Speaker)
+        if view_kwargs.get('event_id'):
+            query_ = query_.join(Event).filter(Event.id == view_kwargs['event_id'])
+        return query_
+
+    def before_create_object(self, data, view_kwargs):
+        if view_kwargs.get('event_id'):
+            event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
+            data['event_id'] = event.id
+
+    decorators = (jwt_required,)
+    schema = SpeakerSchema
+    data_layer = {'session': db.session,
+                  'model': Speaker,
+                  'methods': {
+                      'query': query,
+                      'before_create_object': before_create_object
+                  }}
+
+
+class SpeakerRelationship(ResourceRelationship):
+    decorators = (jwt_required,)
+    schema = SpeakerSchema
+    data_layer = {'session': db.session,
+                  'model': Speaker}
+
+
+class SpeakerDetail(ResourceDetail):
```

Comment:
put this above speakerRelationship, please follow the code style as the rest of the project